Tasks/Grunt/grunttask.ts

```diff
@@ -2,50 +2,121 @@
 
 import path = require('path');
 import tl = require('vsts-task-lib/task');
+import fs = require('fs');
+import os = require('os');
 
-tl.setResourcePath(path.join( __dirname, 'task.json'));
+tl.setResourcePath(path.join(__dirname, 'task.json'));
 
 var gruntFile = tl.getPathInput('gruntFile', true, true);
-
+var grunt = tl.which('grunt', false);
+var isCodeCoverageEnabled = tl.getBoolInput('enableCodeCoverage');
+var publishJUnitResults = tl.getBoolInput('publishJUnitResults');
+var testResultsFiles = tl.getInput('testResultsFiles', publishJUnitResults);
 var cwd = tl.getPathInput('cwd', true, false);
 tl.mkdirP(cwd);
 tl.cd(cwd);
 
-var grunt = tl.which('grunt', false);
-
 tl.debug('check path : ' + grunt);
-if(!tl.exist(grunt)) {
-	tl.debug('not found global installed grunt-cli, try to find grunt-cli locally.');	
-	var gt = tl.createToolRunner(tl.which('node', true));	
-	
+if (!tl.exist(grunt)) {
+	tl.debug('not found global installed grunt-cli, try to find grunt-cli locally.');
+	var gt = tl.createToolRunner(tl.which('node', true));
 	var gtcli = tl.getInput('gruntCli', true);
 	gtcli = path.resolve(cwd, gtcli);
-	
 	tl.debug('check path : ' + gtcli);
-	if(!tl.exist(gtcli)) {
+	if (!tl.exist(gtcli)) {
 		tl.setResult(tl.TaskResult.Failed, tl.loc('GruntCliNotInstalled', gtcli));
 	}
-	
 	gt.pathArg(gtcli);
 }
 else {
 	var gt = tl.createToolRunner(grunt);
 }
 
+if (isCodeCoverageEnabled) {
+	var npm = tl.createToolRunner(tl.which('npm', true));
+	npm.argString('install istanbul');
+	var testFramework = tl.getInput('testFramework', true);
+	var testSrc = tl.getPathInput('testFiles', true, false);
+	var istanbul = tl.createToolRunner(tl.which('node', true));
+	istanbul.arg('./node_modules/istanbul/lib/cli.js');
+	istanbul.argString('cover --report cobertura --report html');
+	if (testFramework.toLowerCase() == 'jasmine') {
+		var jasmineInit = tl.createToolRunner(tl.which('node', true));
+		jasmineInit.argString('./node_modules/jasmine/bin/jasmine.js init');
+		istanbul.arg('./node_modules/jasmine/bin/jasmine.js');
```

Comment:
I don't think this is required. [http://jasmine.github.io/2.4/node.html](http://jasmine.github.io/2.4/node.html)
It's required only to create JSON where you can configure